### PR TITLE
[New Rule] Remote File Download via Scripting

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/646

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by Samirbous** [comment]
image

File: rules/windows/command_and_control_remote_file_copy_scripts.toml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+[metadata]
+creation_date = "2020/11/29"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/11/29"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies cscript.exe or wscript.exe being used to download an executable file from a remote destination.
```
Comment (bm11100) [inline rules/windows/command_and_control_remote_file_copy_scripts.toml:10]:
```suggestion
Identifies built-in Windows script interpreters (cscript.exe or wscript.exe) being used to download an executable file from a remote destination.
```

Comment (bm11100) [reply rules/windows/command_and_control_remote_file_copy_scripts.toml:10]: What about other interpreters like cmd, powershell / pwsh? Would they be useful additions?

Comment (Samirbous) [reply rules/windows/command_and_control_remote_file_copy_scripts.toml]: different noise level, w|cscript are less verbose, powershell is captured by another PR.

File: rules/windows/command_and_control_remote_file_copy_scripts.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+[rule]
+author = ["Elastic"]
+description = """
+Identifies built-in Windows script interpreters (cscript.exe or wscript.exe) being used to download an executable file from a remote destination.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Remote File Download via Scripting"
```
Comment (bm11100) [inline rules/windows/command_and_control_remote_file_copy_scripts.toml:16]:
```suggestion
name = "Remote File Download via Script Interpreter"
```

**Review by bm11100 — APPROVED** [review]
🚣

File: rules/windows/command_and_control_remote_file_copy_scripts.toml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+[metadata]
+creation_date = "2020/11/29"
+maturity = "production"
+updated_date = "2020/11/29"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies built-in Windows script interpreters (cscript.exe or wscript.exe) being used to download an executable file from a remote destination.
```
Comment (brokensound77) [inline rules/windows/command_and_control_remote_file_copy_scripts.toml:9]: linted?

File: rules/windows/command_and_control_remote_file_copy_scripts.toml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Command and Control"]
+type = "eql"
+
+query = '''
+sequence by host.id, process.entity_id with maxspan = 30s
+  [network where process.name : ("wscript.exe", "cscript.exe") and network.protocol != "dns" and
+   network.direction == "outgoing" and network.type == "ipv4" and destination.ip !="127.0.0.1"
+  ]
+  [file where event.type == "creation" and file.extension : ("exe", "dll")]
```
Comment (brokensound77) [inline rules/windows/command_and_control_remote_file_copy_scripts.toml:27]: nit:

```suggestion
sequence by host.id, process.entity_id with maxspan=30s
  [network where process.name : ("wscript.exe", "cscript.exe") and network.protocol != "dns" and
   network.direction == "outgoing" and network.type == "ipv4" and destination.ip != "127.0.0.1"
  ]
  [file where event.type == "creation" and file.extension : ("exe", "dll")]
```

Comment (brokensound77) [reply rules/windows/command_and_control_remote_file_copy_scripts.toml:27]: May not need the `maxspan`, especially for long running scripts

**Review by brokensound77 — APPROVED** [review]
A couple comments - LGTM 👍

File: rules/windows/command_and_control_remote_file_copy_scripts.toml (hunk 4 above)
Comment (Samirbous) [reply rules/windows/command_and_control_remote_file_copy_scripts.toml:27]: good idea

File: rules/windows/command_and_control_remote_file_copy_scripts.toml (hunk 3 above)
Comment (Samirbous) [reply rules/windows/command_and_control_remote_file_copy_scripts.toml]: done

**Pull request merged**